src/components/Bar.jsx

```diff
@@ -77,46 +79,58 @@ class Bar extends Component {
     const { usageTracker, claudyOpened,
-      enableClaudy, drawerVisible, fireClaudy, displaySupport } = this.state
+      enableClaudy, fireClaudy, onDrawer, displayOnMobile, isPublic } = this.props
+    const { drawerVisible } = this.store
+    return (__TARGET__ !== 'mobile' || displayOnMobile) && !isPublic ? <div className='coz-bar-flex-container' key='nav'>
+      <Drawer visible={drawerVisible} onClose={this.toggleDrawer} onClaudy={(enableClaudy && (() => this.toggleClaudy(true))) || false} isClaudyLoading={fireClaudy} drawerListener={() => onDrawer(this.state.drawerVisible)} toggleSupport={this.toggleSupport} />
+      <Nav toggleSupport={this.toggleSupport} />
+      {enableClaudy &&
+        <Claudy
+          usageTracker={usageTracker}
+          fireClaudy={fireClaudy}
+          onToggle={() => this.toggleClaudy(false)}
+          opened={claudyOpened}
+        />
+      }
+    </div> : null
+  }
+
+  render () {
+    const { t } = this.props
+    const { fireClaudy, displaySupport } = this.state
+    const { barLeft, barRight, barCenter } = this.store
     return (
       <div className='coz-bar-container'>
-        <h1 lang={lang} className={`coz-bar-title ${replaceTitleOnMobile ? 'coz-bar-hide-sm' : ''}`}>
-          <img className='coz-bar-hide-sm' src={iconPath} width='32' />
-          {appEditor && <span className='coz-bar-hide-sm'>{appEditor} </span>}
-          <strong>{appName}</strong>
-          <sup className='coz-bar-hide-sm coz-bar-beta-status'>{t('beta')}</sup>
-        </h1>
-        <hr className='coz-sep-flex' />
-        {(__TARGET__ !== 'mobile' || displayOnMobile) && !isPublic &&
-          <div className='coz-bar-flex-container'>
-            <button className='coz-bar-burger' onClick={this.toggleDrawer} data-icon='icon-apps'>
-              <span className='coz-bar-hidden'>{t('drawer')}</span>
-            </button>
-            <Drawer visible={drawerVisible} onClose={this.toggleDrawer} onClaudy={(enableClaudy && (() => this.toggleClaudy(true))) || false} isClaudyLoading={fireClaudy} drawerListener={() => onDrawer(this.state.drawerVisible)} toggleSupport={this.toggleSupport} />
-            <Nav toggleSupport={this.toggleSupport} />
-            {enableClaudy &&
-              <Claudy
-                usageTracker={usageTracker}
-                fireClaudy={fireClaudy}
-                onToggle={() => this.toggleClaudy(false)}
-                opened={claudyOpened}
-              />
-            }
-          </div>
-        }
-        {displaySupport &&
-          <SupportModal
-            onClose={this.toggleSupport}
-          />
-        }
+        { barLeft || this.renderLeft() }
+        { barCenter || this.renderCenter() }
+        <hr className='coz-sep-flex' key='separator'/>
+        { barRight || this.renderRight() }
+        {displaySupport && <SupportModal onClose={this.toggleSupport} />}
       </div>
     )
   }
 }
 
-export default translate()(Bar)
+export default compose(
```

Comment:
Since `compose` is used by redux' reducers, why did you change the `lodash` function name?

Reply:
`compose` is a generic term that was used before `redux` and exactly maps to the mathematical concept of function composition. IMHO I do not think that `flowRight` is particularly self-explanatory and this is why I prefer to alias this method. Do you prefer `flowRight` ?